src/qibo/backends/__init__.py

```diff
@@ -144,12 +144,8 @@ def _default_transpiler(cls):
             and natives is not None
             and connectivity_edges is not None
         ):
-            # only for q{i} naming
-            node_mapping = {q: i for i, q in enumerate(qubits)}
-            edges = [
-                (node_mapping[e[0]], node_mapping[e[1]]) for e in connectivity_edges
-            ]
-            connectivity = nx.Graph(edges)
+            connectivity = nx.Graph(connectivity_edges)
+            connectivity.add_nodes_from(qubits)
```

Comment:
I put this to insert nodes when #qubits is 1 or if there are remaining unconnected nodes. Do these cases happen?

Reply:
Unconnected nodes are not relevant for the various processes involving the connectivity (i.e. routing and placing), because they do not require nor can be involved in swaps.

So, in those cases, the best is just that you work on top of the given connectivity, and you work out the best permutation/swaps additions.
If you are just playing with swaps, adding them on the involved qubits should leave the unconnected ones unaltered (at most, if a new circuit is being created, you may iterate on them at the end, appending all the gates for the qubits in the original circuits, which are not nodes in the connectivity).
Concerning the permutations, instead, you could compute them restricted to the given connectivity, and then apply them to the original list, instead of creating a new list (which could miss the unconnected elements). E.g. if the final permutation is something like:
```py
pi = {0: 4, 1: 1, 4: 0}
```
on a 5-qubit chip with arbitrary names, you could apply it as follows:
```py
new_circ.wire_names = old_circ.wire_names.copy()
for i, o in pi.items():
    new_circ.wire_names[i] = old_circ.wire_names[o]
```

---

`i` and `o` may be swapped, according to the convention used on the permutation. Moreover, the minimal representation of a permutation would be:
```py
[4, 1, 0]
```
i.e. just the values. And you would obtain back the case above by `zip`ping them with the keys obtained from the (sorted) graph nodes.